consul/consul.go

```diff
@@ -125,6 +131,35 @@ func (r *ConsulAdapter) buildCheck(service *bridge.Service) *consulapi.AgentServ
 	return check
 }
 
+func (r *ConsulAdapter) PostAttributes(service *bridge.Service) error {
+	for k, v := range service.Attrs {
+		data := new(consulapi.KVPair)
+		data.Key = buildServiceKey(service, k)
+		data.Value = []byte(v)
+		log.Println("Updating Consul KV: ", data.Key, v)
+		_, err := r.client.KV().Put(data, nil)
```

Comment:
We did not use the Consulkv code because we need the actual registered service to be stored as a service instead of a k/v attribute. This enables consul to keep track of multiple distributed instances of a service.